### ISSUE TITLE
Download only columns that are in the question

### ISSUE THREAD

**Issue by mvrabel** [metabase/metabase#5295]
I want to be able 
![all](https://user-images.githubusercontent.com/12628893/26888528-939157a8-4bab-11e7-8be7-e14d7922bf1f.PNG)
![filtered](https://user-images.githubusercontent.com/12628893/26888529-93957432-4bab-11e7-90ab-bf8a9143c081.PNG)

to download only those columns that I have filtered in the 'custozmize table' dialog.

For example: 
I have a table with 5 columns : 
name, age, city, mother_name, father_name
I filter out:  'mother_name' and 'father_name' .
The result table in the dashboard will contain only 3 columns 'name', 'age', 'city'. 

But when I download xlsx the downloaded file will have all 5 columns.

We have one big table at work and I want to be able to create report from only a few columns.

- browser and the version:  58.0.3029.110 (64-bit)
- operating system: Win 10
- databases: Postgressql 9.6
- Metabase version: v0.24.2 Built on 2017-06-01
- Metabase hosting environment: CentOS 7
- Metabase internal database:  I think H2 (default), I only downloaded the jar and conected to the database.

Thanks, 
Martin

**Comment by Varunram** [comment]
@mvrabel I do agree that exporting the table as is would be really good. AFAIK, this issue mostly occurs with pivoted tables and in some `GROUP_BY` clauses. Not exactly sure how we would export as-is though, given that the results from the sql queries are exported into `.xlsx` files. Waiting for others' views on this

**Comment by mvrabel** [comment]
@Varunram I don't use GROUP BY. I just have one big table and I made some columns invisible and reordered some columns.
If the result is directly exported into xlsx then I can solve it by creating custom query. Thanks!
But It would be awesome if you could add this feature.

**Comment by salsakran** [comment]
Are you saving the question after you make the changes to the visible columns?

When downloading the results of a question, we re-run the query (and include all rows instead of capping them).

**Comment by mvrabel** [comment]
Yes, I saved the question before downloading and it still downloaded all columns.

**Comment by ujjwal-gupta** [comment]
Not only that it even includes columns that have been marked as "show only in detailed view" from Data Model in Admin Panel in the exported file which exposes a lot of undesirable technical stuff to users. I think that the priority should be increased considering this.

**Comment by ryanmccoy** [comment]
Modifying the export feature to at least include the option to download in the exact format shown in the results panel of a new question would be amazing. 

I assume that most people doing business intelligence or general data analytics are going to use a plethora of tools, each dependent on the situation at hand. 

For example: I spend a lot of time massaging data in MetaBase and finally end up with a results table perfectly formatted, but this format still needs to be summarised into some simple pivot tables. Often times to do this would be quicker in a spreadsheet than trying to go back through the query builder and format the sql statement even further in order to essentially build the summarised data with SQL itself. However when I export, the results are fed to me exactly as they would be from the sql statement run in a normal sql browser, which means that I am now forced to make a choice of formatting the raw results again OR trying to construct the necessary sql procedures to do so within metabase. Which is an unfortunate limitation of options when my ability to choose the tool for the given situation is paramount to being able to execute my analysis in a timely fashion.

**Comment by annisun** [comment]
If I have two tables associated, can I export the information from the selected column to the associated table?

**Comment by tashiritzy** [comment]
If you use Grouped By option instead of selecting the fields from the check list, it serves some purpose.

**Comment by PauloHFM** [comment]
I have a problem, not about the visible or invisible columns, but I saved the question in some column order to create a report for downloading in xlsx. 

The question is a join of 3 tables made in custom question option. The downloaded file doesn't represent the order i put on the question! 

There is a way of doing this? without changing the tables, or the metadata in the the data model?

Some issue about this?

**Comment by flamber** [comment]
@PauloHFM Currently no formatting settings are parsed to exports, but issue #10879 covers that

**Comment by PauloHFM** [comment]
@flamber  Thanks !
